mite_selenium/__init__.py

```diff
@@ -55,30 +55,43 @@ def stop(self):
 def send_selenium_times(context):
     """Maybe would probably need more error handling
     if somebody tried to use with an incompatible browser"""
-    if context.browser.capabilities['browserName'] != 'chrome':
+    if context.browser.capabilities["browserName"] != "chrome":
         logger.warning(
-            'Timing metrics in %s are not currently supported',
-            context.browser.capabilities['browserName'],
+            "Timing metrics in %s are not currently supported",
+            context.browser.capabilities["browserName"],
         )
         return
 
     timings = context.browser.execute_script(
         'return performance.getEntriesByType("navigation")'
     )[0]
+    request_response_time = timings["responseEnd"] - timings["requestStart"]
+    first_contentful_paint = context.browser.execute_script(
+        'return performance.getEntriesByName("first-contentful-paint")'
+    )[0]["startTime"]
+    time_to_interactive = timings["domComplete"] - first_contentful_paint
+
     context.send(
         "http_selenium_metrics",
-        dom_complete=timings["domComplete"],
-        dom_interactive=timings["domInteractive"],
-        transfer_size=timings["transferSize"],
-        transfer_start_timings=["requestStart"],
-        tls_time=timings["secureConnectionStart"],
-        dns_time=timings["domainLookupEnd"],
-        connect_time=timings["connectEnd"],
-        total_time=timings["duration"],
+        dom_complete=_convert_ms_to_seconds(timings["domComplete"]),
+        dom_interactive=_convert_ms_to_seconds(timings["domInteractive"]),
+        transfer_size=_convert_ms_to_seconds(timings["transferSize"]),
+        transfer_start_timings=_convert_ms_to_seconds(timings["requestStart"]),
+        tls_time=_convert_ms_to_seconds(timings["secureConnectionStart"]),
```

Comment:
I'm not totally sure what to do about these two (but they are not correct as written -- they should be somethingEnd - somethingStart in general).  There are three metrics, connectStart, connectEnd, and secureConnectionStart.  I think the way it breaks down is that connectStart until secureConnectionStart is the TCP connection overhead (`tcp_time` maybe), and secureConnectionStart to connectEnd is the TLS overhead (`tls_time`).  But it might be the case that the connection is not TLS (which we can get by inspecting timings["name"]), in which case we should just use connectStart - connectEnd and set the tls_time to zero.

There are also some exotic new protocols (http/3, sometimes also called SPDY or QUIC) that frontload some of the TLS negotiation overhead into the layer4 negotiation (which is not TCP any more, but a new protocol).  As long as we're talking directly to boring http servers inside the platform, then this will work fine.  But if we start talking to e.g. akamai, then (as they upgrade their systems) the browser might autonegotiate something exciting.  We can check nextHopProtocol and make sure it's equal to http/1.1 -- if not we probably want to log a warning that the more detailed assumptions about how the timing breaks down might not be valid.